tests/test_feedexport.py

```diff
@@ -606,6 +607,62 @@ class MyItem2(scrapy.Item):
         yield self.assertExported(items, header, rows,
                                   settings=settings, ordered=True)
 
+        # fields may be defined as a comma-separated list
+        header = ["foo", "baz", "hello"]
+        settings = {'FEED_EXPORT_FIELDS': ",".join(header)}
+        rows = [
+            {'foo': 'bar1', 'baz': '',      'hello': ''},
+            {'foo': 'bar2', 'baz': '',      'hello': 'world2'},
+            {'foo': 'bar3', 'baz': 'quux3', 'hello': ''},
+            {'foo': '',     'baz': '',      'hello': 'world4'},
+        ]
+        yield self.assertExported(items, header, rows,
+                                  settings=settings, ordered=True)
+
+        # fields may also be defined as a JSON array
+        header = ["foo", "baz", "hello"]
+        settings = {'FEED_EXPORT_FIELDS': json.dumps(header)}
+        rows = [
+            {'foo': 'bar1', 'baz': '',      'hello': ''},
+            {'foo': 'bar2', 'baz': '',      'hello': 'world2'},
+            {'foo': 'bar3', 'baz': 'quux3', 'hello': ''},
+            {'foo': '',     'baz': '',      'hello': 'world4'},
+        ]
+        yield self.assertExported(items, header, rows,
+                                  settings=settings, ordered=True)
+
+        # custom output field names can be specified
+        header = OrderedDict((
+            ("foo", "Foo"),
+            ("baz", "Baz"),
+            ("hello", "Hello"),
+        ))
```

Comment:
Could you please add a test for a regular dict, and ensure that order is preserved in Python 3.7+?